### PR TITLE
Feature/more flexible format str

### PR DESCRIPTION
**Pull request by prescod**
Internal reorganization of this code.

Allow callers to pass arbitrary keyword arguments to this Jinja function.

Make the function more globally visible by moving the file

### CODE REVIEW

**Comment by prescod** [comment]
@davisagli I've double checked that there are no references in Cumulus. All such references go through the Robot library which is updated here.

I also added another feature today, which will be useful in my other project. The ability to generate e.g. Spanish-sounding names.

File: cumulusci/core/template_utils.py [hunk 1]
```diff
@@ -47,9 +48,11 @@ def __getattr__(self, name):
 
 faker_template_library = FakerTemplateLibrary()
 
+Template = lru_cache(512)(Template)
+
 
-def format_str(value, i):
-    if isinstance(value, str):
-        value = Template(value).render(number=i, fake=faker_template_library)
+def format_str(value, variables={}, fake=faker_template_library):
```
Comment (davisagli) [inline cumulusci/core/template_utils.py]: Don't use mutable defaults! It's a trap!

Comment (prescod) [reply cumulusci/core/template_utils.py]: Good catch!

Comment (prescod) [reply cumulusci/core/template_utils.py]: Although it is a trap, in this case the function doesn't (and shouldn't) mutate the dict. In the future I would use a frozenmap as the default to enforce that.

**Pull request merged**